File: androidApp/src/main/java/com/mirego/kmp/boilerplate/views/components/ButtonView.kt
package com.mirego.kmp.boilerplate.views.components

import androidx.compose.material.Button
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import com.mirego.kmp.boilerplate.viewmodels.components.ButtonViewModel

@Composable
fun ButtonView(viewModel: ButtonViewModel) {
    val state: ButtonViewModel.State by viewModel.state.collectAsState()

    Button(viewModel::action, enabled = state.enabled) {
        state.text?.let { Text(it) }
    }
}

File: ios/iosApp/Views/Components/ButtonView.swift
import Shared
import SwiftUI

struct ButtonView: View {

    private let viewModel: ButtonViewModel

    @State private var state: ButtonViewModelState

    init(_ viewModel: ButtonViewModel) {
        self.viewModel = viewModel
        self.state = viewModel.state.value
    }

    var body: some View {
        Button(state.text ?? "", action: viewModel.action)
            .disabled(!state.enabled)
            .task {
                for await state in viewModel.state {
                    self.state = state
                }
            }
    }
}
